Solutions/Corvus.Json.ExtendedTypes/Corvus.Json/JsonAny.cs

```diff
@@ -1096,6 +1096,72 @@ public JsonAny Add<TItem>(TItem item)
             return this;
         }
 
+        /// <inheritdoc/>
+        public JsonAny Add<TItem1, TItem2>(TItem1 item1, TItem2 item2)
+            where TItem1 : struct, IJsonValue
+            where TItem2 : struct, IJsonValue
+        {
+            if (this.ValueKind == JsonValueKind.Array || this.ValueKind == JsonValueKind.Undefined)
+            {
+            return this.AsArray.Add(item1, item2);
+        }
+
+            return this;
```

Comment:
So this is interesting - the behaviour for "remove" and "replace" (and all the TryGets, Has etc.) is correct.

Are these misnamed (should be TryAdd?) or should they throw?

I don't like TryAdd() because the `bool, out` pattern is horrible for chaining them together.

So I think that these *should* throw an `InvalidOperationException`

Reply:
(The "Remove" behaviour is consistent with that of ImmutableDictionary, which is why propose leaving that as-is).

Reply:
And now I'm worried about suddenly injecting exception-throwing behaviour into a path which is all exception-free with "validation when you're done".

Reply:
I think I will leave this like this and we can open an issue for a breaking change if you would like to consider it!

Reply:
I've created #31 to track this.